models/enterprise_sonic/snmp/sonic_snmp.yml

```diff
@@ -2,52 +2,284 @@
+    description:
+      - This module provides configuration management of SNMP for devices running SONiC
+    author: Aida Shumburo (@aida-shumburo)
+    options:
+      config:
+        description:
+          - SNMP configuration
+        type: dict
+        suboptions:
+          agentaddress:
+            description:
+              - List of SNMP agent address configuration
+              - I(interface) and I(vrf) are mutually exclusive.
+            type: list
+            elements: dict
+            suboptions:
+              ip:
+                description:
+                  - IPv4 or IPv6 address
+                type: str
+                required: true
+              port:
+                description:
+                  - UDP listening port, range 1024-65535
+                type: int
+              interface:
+                description:
+                  - Name of interface
+                type: str
+              vrf:
+                description:
+                  - Name of VRF
+                type: str
+          community:
+            description:
+              - List of SNMP community configuration
+            type: list
+            elements: dict
+            suboptions:
+              group:
+                description:
+                  - Name of community group
+                type: str
+              name:
+                description:
+                  - Name of community
+                type: str
+                required: true
+          engine:
+            description:
+              - SNMP engine ID
+              - Octet (hex) string with colon separated, 5 to 32 octets
+            type: str
+          user:
+            description:
+              - List of SNMP user configuration
+            type: list
+            elements: dict
+            suboption:
+              group:
+                description:
+                  - Name of user group
+                type: str
+              name:
+                description:
+                  - Name of user
+                type: str
+                required: true
+              auth:
+                description:
+                  - User authentication configuration
+                type: dict
+                suboptions:
+                  auth_type:
+                    description:
+                      - Authentication type
+                    type: str
+                    choices:
+                      - md5
+                      - sha
+                  key: 
+                    description: 
+                      - Authentication key
+                    type: str
+              priv:
+                description:
+                  - Privacy configuration
+                type: dict
+                suboptions:
+                  priv_type:
+                    description:
+                      - Type of encryption
+                      - Choices are 'aes' or 'des' 
+                    type: str
```

Comment:
As with the "auth_type" option, please add a "choices" attribute here instead of only stating it in the description. This will eliminate the need to check this in the executable code.

(The description part for the choices is optional if an explicit "choices" attribute is used.)

Reply:
Verified.